### PR TITLE
Feature/cfisher/sidebar toggle

### PR DESCRIPTION
**Pull request by cehfisher**
Added accessible button toggle element that expands/hides the sidebar with mouse click or keyboard navigation. Just tab to the button and press space bar or enter to toggle.

Added `visuallyhidden ` selector to hide the toggle button name from visual users, but will be read for screen readers (so those users know what they are toggling). This selector can be used for other similar elements or to give more context to links (for example: http://a11y-style-guide.com/style-guide/section-general.html#kssref-general-read-more). 

For the actual style of the toggle, I created something simple to go with the rest of your style guide ascetic, but update what you need to. Also, I did some quick tests on Chrome, Safari, and FireFox (latest versions), but might need some deeper cross-browser testing.

### CODE REVIEW

File: _includes/primary-nav-toggle.html [hunk 1]
```diff
@@ -0,0 +1,15 @@
+<button type="button" class="nav-btn c-primary-nav__item">
+  <span class="visuallyhidden">
+    Primary Navigation Toggle
+  </span>
+  <span class="toggle-open">
+    <svg class="c-icon c-primary-nav__icon" aria-hidden="true">
```
Comment (Ryuno-Ki) [inline _includes/primary-nav-toggle.html]: Great to see someone using ARIA!

Just some suggestions:
1. Add `aria-controls` to help accessibility tools („AT”) understand the relationship between the button and the element it affects.
2. Add a `<title>` as first child of `<svg>` (+ aria-label), so AT can read out, what is displayed on the icon (assuming it is not used ornamentally).

Comment (cehfisher) [inline _includes/primary-nav-toggle.html]: thanks for the feedback! I can look into updating #1. For #2 the icon is ornamental so I purposely left out the alt info so the AT can skip it.

**Comment by Ryuno-Ki** [comment]
Look at spec of [aria-expanded](https://www.w3.org/TR/wai-aria/states_and_properties#aria-expanded) and [aria-controls](https://www.w3.org/TR/wai-aria/states_and_properties#aria-controls) for semantics. I don't think, you need to add [role definitions](https://www.w3.org/TR/wai-aria/roles#abstract_roles) here... W3C didn't used it [in its example] (https://www.w3.org/WAI/GL/wiki/Using_aria-expanded_to_indicate_the_state_of_a_collapsible_element) at least...

**Comment by bradfrost** [comment]
@cehfisher Thanks so much for this, and forgive the delayed response in getting back to you.

The functionality looks fantastic and of course pleased to see the focus on accessibility! I think with a bit of a style tune-up to align sidebar and main body animation, we should be good to merge this in. Thanks again!

**Comment by cehfisher** [comment]
@bradfrost no worries! not sure what you wanted for the style, so guessed a bit. what specific changes do you need at this point?

**Comment by bradfrost** [comment]
> what specific changes do you need at this point?

@cehfisher RIght now, the main content snaps into place while the sidebar transitions:

drawer

 If we can get the content to transition along with the sidebar, that would feel better! Thanks a ton for your help on this.

**Comment by cehfisher** [comment]
OK added a transition for the body content, so it is less jerky. due to the differences in starting point vs ending point (sidebar has more distance to cover vs main content section), not sure you can ever line it up 100% perfectly but this gets it closer. let me know if you need any more changes and happy 4th :)
slide-action

**Comment by cehfisher** [comment]
@bradfrost are there any other changes you need at this time?

**Comment by bradfrost** [comment]
@cehfisher I'm finally getting a chance to look at the code for this functionality, and it looks like there will need to be some significant refactoring to match the syntax and conventions (which roughly match [this](http://bradfrost.com/blog/post/css-architecture-for-design-systems/)) used throughout the site. For instance, there is a `<span class="visuallyhidden">`, but a utility class of `u-is-vishidden` already exists. 

I'll update the syntax and then it should be good to merge.

**Comment by cehfisher** [comment]
OK sounds like it's best to delete this PR then. Thanks for the info!

**Comment by cehfisher** [comment]
if it makes sense @bradfrost i'd like to try and see if i can match the syntax/conventions with a new PR. i've been working in FE dev for awhile now, but we all have our coding nuances. i'd like to read your post and see if i can make this PR more clean. if you have any additional hints, let me know. thanks!

**Comment by bradfrost** [comment]
@cehfisher Feel free to take a stab at this, as my time is really limited. I'd suggest looking at the rest of the codebase to see how CSS and JavaScript conventions are structured.

**Pull request closed without merging**